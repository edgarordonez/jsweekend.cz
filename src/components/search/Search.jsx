import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import Form from 'muicss/lib/react/form';
import Input from 'muicss/lib/react/input';
import Button from 'muicss/lib/react/button';

export default class Search extends PureComponent {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    const fields = {
      from: this.from.value,
      to: this.to.value,
      date: this.date.value,
    };

    this.props.handleSubmit(fields);
  }

  render() {
    return (
      <Fragment>
        <Form
          inline
          onSubmit={this.handleSubmit}
        >
          <legend>Flight search</legend>
          <Input
            type="text"
            label="From"
            placeholder="From"
            required
            ref={({ controlEl }) => this.from = controlEl}
          />
          <Input
            type="text"
            label="To"
            placeholder="To"
            required
            ref={({ controlEl }) => this.to = controlEl}
          />
          <Input
            type="date"
            label="Date"
            required
            ref={({ controlEl }) => this.date = controlEl}
          />
          <Button variant="raised">Submit</Button>
        </Form>
      </Fragment>
    );
  }
}

Search.propTypes = {
  handleSubmit: PropTypes.func,
};
