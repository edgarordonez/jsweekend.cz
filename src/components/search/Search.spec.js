import React from 'react';
import { shallow } from 'enzyme';
import { Search } from './Search';

it('renders without crashing', () => {
  const data = {
    handleSubmit: (event) => event,
    client: null
  };
  const search = shallow(<Search handleSubmit={data.handleSubmit} client={data.client} />);

  expect(search).toMatchSnapshot();
});
