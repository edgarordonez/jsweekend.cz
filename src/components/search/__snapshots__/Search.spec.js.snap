// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Form
        className=""
        inline={true}
        onSubmit={[Function]}
>
        <legend>
                Flight search
        </legend>
        <_class
                className=""
                floatingLabel={false}
                hint={null}
                invalid={false}
                label="From"
                placeholder="From"
                required={true}
                type="text"
        />
        <_class
                className=""
                floatingLabel={false}
                hint={null}
                invalid={false}
                label="To"
                placeholder="To"
                required={true}
                type="text"
        />
        <_class
                className=""
                floatingLabel={false}
                hint={null}
                invalid={false}
                label="Date"
                required={true}
                type="date"
        />
        <Button
                className=""
                color="default"
                size="default"
                variant="raised"
        >
                Submit
        </Button>
</Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <legend>
            Flight search
</legend>,
          <_class
            className=""
            floatingLabel={false}
            hint={null}
            invalid={false}
            label="From"
            placeholder="From"
            required={true}
            type="text"
/>,
          <_class
            className=""
            floatingLabel={false}
            hint={null}
            invalid={false}
            label="To"
            placeholder="To"
            required={true}
            type="text"
/>,
          <_class
            className=""
            floatingLabel={false}
            hint={null}
            invalid={false}
            label="Date"
            required={true}
            type="date"
/>,
          <Button
            className=""
            color="default"
            size="default"
            variant="raised"
>
            Submit
</Button>,
        ],
        "className": "",
        "inline": true,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Flight search",
          },
          "ref": null,
          "rendered": "Flight search",
          "type": "legend",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "floatingLabel": false,
            "hint": null,
            "invalid": false,
            "label": "From",
            "placeholder": "From",
            "required": true,
            "type": "text",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "floatingLabel": false,
            "hint": null,
            "invalid": false,
            "label": "To",
            "placeholder": "To",
            "required": true,
            "type": "text",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "floatingLabel": false,
            "hint": null,
            "invalid": false,
            "label": "Date",
            "required": true,
            "type": "date",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Submit",
            "className": "",
            "color": "default",
            "size": "default",
            "variant": "raised",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Form
          className=""
          inline={true}
          onSubmit={[Function]}
>
          <legend>
                    Flight search
          </legend>
          <_class
                    className=""
                    floatingLabel={false}
                    hint={null}
                    invalid={false}
                    label="From"
                    placeholder="From"
                    required={true}
                    type="text"
          />
          <_class
                    className=""
                    floatingLabel={false}
                    hint={null}
                    invalid={false}
                    label="To"
                    placeholder="To"
                    required={true}
                    type="text"
          />
          <_class
                    className=""
                    floatingLabel={false}
                    hint={null}
                    invalid={false}
                    label="Date"
                    required={true}
                    type="date"
          />
          <Button
                    className=""
                    color="default"
                    size="default"
                    variant="raised"
          >
                    Submit
          </Button>
</Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <legend>
              Flight search
</legend>,
            <_class
              className=""
              floatingLabel={false}
              hint={null}
              invalid={false}
              label="From"
              placeholder="From"
              required={true}
              type="text"
/>,
            <_class
              className=""
              floatingLabel={false}
              hint={null}
              invalid={false}
              label="To"
              placeholder="To"
              required={true}
              type="text"
/>,
            <_class
              className=""
              floatingLabel={false}
              hint={null}
              invalid={false}
              label="Date"
              required={true}
              type="date"
/>,
            <Button
              className=""
              color="default"
              size="default"
              variant="raised"
>
              Submit
</Button>,
          ],
          "className": "",
          "inline": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Flight search",
            },
            "ref": null,
            "rendered": "Flight search",
            "type": "legend",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "floatingLabel": false,
              "hint": null,
              "invalid": false,
              "label": "From",
              "placeholder": "From",
              "required": true,
              "type": "text",
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "floatingLabel": false,
              "hint": null,
              "invalid": false,
              "label": "To",
              "placeholder": "To",
              "required": true,
              "type": "text",
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "floatingLabel": false,
              "hint": null,
              "invalid": false,
              "label": "Date",
              "required": true,
              "type": "date",
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Submit",
              "className": "",
              "color": "default",
              "size": "default",
              "variant": "raised",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
